gpsrenda/bin/renda

```diff
@@ -0,0 +1,89 @@
+#!/usr/bin/env python
+import argparse as ap
+from datetime import datetime
+from datetime import timedelta
+from glob import glob
+from importlib import import_module
+from os import makedirs
+from os.path import basename, dirname, join
+
+import cairo
+import moviepy.editor as mpy
+import numpy as np
+import yaml
+
+from gpsrenda.datasources import FitDataSource
+from gpsrenda.utils import extract_start_time, timestamp_to_seconds
+
+def renda(video_paths, data_path, time_offset, config_path, preview):
+    data_source = FitDataSource(data_path)
+
+    with open(config_path, 'r') as config_file:
+        config_data = yaml.safe_load(config_file)
+
+    widgets = []
+    widget_module = import_module('gpsrenda.widgets.widgets')
+    for widget_spec in config_data['widgets']:
+        widget_type = widget_spec.pop('type')
+        widget_class = getattr(widget_module, widget_type+'Widget')
+        widget = widget_class(data_source=data_source, **widget_spec)
+        widgets.append(widget)
+
+    for video_path in video_paths:
+        clip = mpy.VideoFileClip(video_path)
+        start_t = timestamp_to_seconds(extract_start_time(video_path))
+
+        def make_frame(t):
+            frame = clip.get_frame(t)
+            h, w = frame.shape[:2]
+            alpha = np.zeros((h, w, 1), dtype=frame.dtype)
```

Comment:
I am surprised that this works, I would have expected that you need the alpha channel to be full of ones

Reply:
I think it works because I strip the alpha channel away before passing it back off to moviepy for encoding anyway. Cairo just needs a HxWx4 array to write to before the alpha channel is dropped.